extendedcopy.go

```diff
@@ -195,40 +212,7 @@ func (opts *ExtendedCopyGraphOptions) FilterAnnotation(key string, regex *regexp
 			return nil, err
 		}
 		var filtered []ocispec.Descriptor
-		for _, p := range predecessors {
-			if p.Annotations == nil {
-				switch p.MediaType {
-				case docker.MediaTypeManifest, ocispec.MediaTypeImageManifest,
-					docker.MediaTypeManifestList, ocispec.MediaTypeImageIndex,
-					artifactspec.MediaTypeArtifactManifest:
-					if err = func() error {
-						rc, err := src.Fetch(ctx, p)
-						if err != nil {
-							return err
-						}
-						defer rc.Close()
-						var manifest struct {
-							Annotations map[string]string `json:"annotations"`
-						}
-						if err := json.NewDecoder(rc).Decode(&manifest); err != nil {
-							return err
-						}
-						if manifest.Annotations == nil {
-							p.Annotations = map[string]string{}
-						} else {
-							p.Annotations = manifest.Annotations
-						}
-						return nil
-					}(); err != nil {
-						return nil, err
-					}
-				}
-			}
```

Comment:
We still need this, don't we?

Reply:
According to the pr description, now the referrer API must return the annotations **if the manifest has annotations**. So we don't need this if we count on that.

Reply:
But I think maybe we still need a `if p.Annotations == nil { p.Annotations = map[string]string{} }` to handle the case that the original manifest doesn't have annotations.

Reply:
Or maybe we can keep this for `src`s that are not a `registry.ReferrerFinder`.

Reply:
Added the decoding back for src that is not a `registry.ReferrerFinder`.